rfcs/0001-masonry-backend.md

```diff
@@ -0,0 +1,166 @@
+
+### Jumping to Masonry
+
+By using the Masonry codebase, the Xilem project will move towards concept Masonry was implementing that Xilem maintainers weren't necessarily considering for Xilem.
+
+Some PRs intended for Xilem may become obsolete once the move to Masonry is complete, because Masonry will go in a different direction from the one this PRs were made for.
+
+
+## Rationale and alternatives
+
+### Separating at the crate level vs the repository level
+
+The initial suggestion was to use Masonry as it currently exists, as a separate repository.
+
+However, people have brought up that it would make the Xilem development process a lot more complex. Features that reached across crates (`xilem_core`, `xilem_native`, `masonry`) would need to be rolled out at coordinated pull requests between repositories.
+
+### Separating at the crate level vs the module level
+
+Masonry and Xilem's code could simply be merged in a single crate, under different modules.
+
+The consensus was mostly that having Masonry as a separate crate would help keep Xilem "honest" and enforce separation of concerns.
+
+More importantly, Masonry was built as a potential foundation for *multiple* native GUI crates, and it would be valuable to conserve that; therefore, while Masonry would still depend on crates from the Linebender ecosystem, it shouldn't be too tied to Xilem.
+
+### Porting Xilem to Masonry vs Masonry to Xilem
+
+In any merge between two projects, one has to be the base the other is grafted on.
+
+This RFC was written under the assumption Xilem would move towards Masonry, but the opposite could be possible.
+
+That said, the Motivation section explains why Masonry is a healthier codebase to start from.
+
+## Unresolved questions
+
+### Do we want to keep the Masonry brand?
+
+The name Masonry was chosen when that crate wasn't officially part of the Linebender project. Even if we want the codebase and the crate to be added to the Linebender umbrella, its name and general branding are up in the air.
+
+Arguments for:
+
+- "Masonry" is catchy name, easy to pronounce in most european languages.
+- The name already exists and the Rust community is vaguely aware of it, to the point is comes up (rarely) in discussions.
+- The name evokes brickwork and building stuff out of sturdy rectangle things. It can evoke having a solid foundation that things are built on top of.
+- I have [the coolest logo idea ever](https://github.com/PoignardAzur/masonry-rs/issues/7).
+
+Arguments against:
+
+- "A masonry layout" is a thing, especially in web development.
+  - People unfamiliar with Rust might come upon this crate and misunderstand what it's for.
+  - When talking about how Masonry does layout, discussion could get confusing.
+
+
+## Future possibilities
+
+Xilem has added features over Druid that Masonry doesn't support, notably virtual lists and async support.
```

Comment:
Well they're prototypes, and as you've already said, they are commented out, and quite a bit out of date/not-working (and I think there may be some possible UX improvements, such as the list callback taking the app state as well as parameter). I think when we're approaching virtualized lists etc. that there will be a refactored version implemented either in xilem as well as a corresponding widget in masonry. A working version is currently on the old [`xilem_tokio`](https://github.com/linebender/druid/tree/xilem_tokio) branch in druid (which is basically the same code as currently in xilem, but working...)

async is an interesting point though, I think long-term async will almost be a requirement for some widgets in masonry. Possibly via a feature gate for async widgets (ideally not hardcoding a specific async runtime)?